src/containers/assignments/AssignmentFormContainer.jsx

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Actions } from 'jumpstate';
+
+import AssignmentForm from '../../components/assignments/AssignmentForm';
+import {
+  ASSIGNMENTS_INITIAL_STATE, ASSIGNMENTS_PROPTYPES
+} from '../../ducks/assignments';
+import {
+  CLASSROOMS_INITIAL_STATE, CLASSROOMS_PROPTYPES
+} from '../../ducks/classrooms';
+import {
+  PROGRAMS_INITIAL_STATE, PROGRAMS_PROPTYPES
+} from '../../ducks/programs';
+
+export class AssignmentFormContainer extends React.Component {
+  constructor() {
+    super();
+
+    this.createAssignment = this.createAssignment.bind(this);
+    this.updateAssignment = this.updateAssignment.bind(this);
+    this.onChange = this.onChange.bind(this);
+    this.onChangeDate = this.onChangeDate.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+  }
+
+  componentWillUnmount() {
+    this.resetFormFields();
+  }
+
+  onChange(event) {
+    if (event.target && event.target.id && event.target.value) {
+      const fields = { ...this.props.formFields, [event.target.id]: event.target.value };
```

Comment:
Not a review comment, but a personal one: _ooh!_  TIL how to use `{ [key_name]: value }` to create on-the-fly keys for {objects}, and that's kinda amazing.

Reply:
One of the great things ES6 gave us!